### PR TITLE
[video-thumbnails][Android] Fix a crash when the source cannot be retrieved

### LINKED ISSUE #27936: expo-video-thumbnails crash on bad uri
### Minimal reproducible example

https://github.com/quick691fr/test-expo-thumbnail

### What platform(s) does this occur on?

Android

### Did you reproduce this issue in a development build?

Yes

### Summary

On incorrect URL (not found, connection reset, ...), getThumbnailAsync() method should reject the promise with error, instead app is crashing with error :
```
java.lang.RuntimeException: setDataSource failed: status = 0x80000000
  android.media.MediaMetadataRetriever._setDataSource(Native Method)
  android.media.MediaMetadataRetriever.setDataSource(MediaMetadataRetriever.java:103)
  expo.modules.videothumbnails.VideoThumbnailsModule$GetThumbnail.execute(VideoThumbnailsModule.kt:85)
  expo.modules.videothumbnails.VideoThumbnailsModule$definition$lambda$4$lambda$2$$inlined$withModuleScope$1.invokeSuspend(VideoThumbnailsModule.kt:114)
  kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
  kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
  kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:42)
  kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
  kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
  kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
  kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
  kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
```

### Environment

```
expo-env-info 1.2.0 environment info:
    System:
      OS: Windows 11 10.0.22631
    Binaries:
      Node: 20.10.0 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.19 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
      npm: 10.5.0 - C:\Program Files\nodejs\npm.CMD
    npmPackages:
      expo: ~50.0.14 => 50.0.14 
      react: 18.2.0 => 18.2.0 
      react-native: 0.73.6 => 0.73.6 
    Expo Workflow: bare
```

### Expo Doctor Diagnostics

```
✔ Check Expo config for common issues
✔ Check package.json for common issues
✔ Check native tooling versions
✔ Check dependencies for packages that should not be installed directly
✔ Check for common project setup issues
✔ Check npm/ yarn versions
✔ Check for issues with metro config
✔ Check Expo config (app.json/ app.config.js) schema
✔ Check for legacy global CLI installed locally
✔ Check that native modules do not use incompatible support packages
✔ Check that packages match versions required by installed Expo SDK
✔ Check that native modules use compatible support package versions for installed Expo SDK
```

### PR DESCRIPTION
**Pull request by mrakesh0608**
# Why

Fixes #27936

# How

1. Added a validation to check valid `sourceFilename`.
2. If the passed `sourceFilename` is valid, but the retriever can't load it due to a bad URI, then it throws an error `setDataSource failed: status = 0x80000000`. To handle this, a catch handler is added.

# Test Plan

Tested using bare-expo.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: apps/native-component-list/src/navigation/ExpoComponentsStackNavigator.tsx [hunk 1]
```diff
@@ -409,6 +409,12 @@ export const Screens: ScreenConfig[] = [
     },
     name: 'Video (expo-video)',
   },
+  {
+    getComponent() {
+      return optionalRequire(() => require('../screens/VideoThumbnailsScreen'));
+    },
+    name: 'Video Thumbnails',
+  },
```
Comment (tsapeta) [inline apps/native-component-list/src/navigation/ExpoComponentsStackNavigator.tsx:417]: `expo-video-thumbnails` is not a component, can you please move it to `ExpoApisStackNavigator` instead?

File: apps/native-component-list/src/screens/ExpoComponentsScreen.tsx [hunk 2]
```diff
@@ -46,6 +46,7 @@ const screens = [
   'Touchables',
   'Video (expo-av)',
   'Video (expo-video)',
+  'Video Thumbnails',
```
Comment (tsapeta) [inline apps/native-component-list/src/screens/ExpoComponentsScreen.tsx:49]: Like above, do it in `ExpoApisScreen.tsx`.

File: apps/native-component-list/src/screens/VideoThumbnailsScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      console.warn(e);
+    }
+  };
+
+  return (
+    <View style={styles.container}>
+      {image && <Image source={{ uri: image }} style={styles.image} />}
+      <Text>{image}</Text>
+      <Button
+        onPress={() => generateThumbnail('http://d23dyxeqlo5psv.cloudfront.net/big_buck_bunny.mp4')}
```
Comment (tsapeta) [inline apps/native-component-list/src/screens/VideoThumbnailsScreen.tsx:24]: Afaik iOS by default is not allowing insecure HTTP connections, so let's use HTTPS
```suggestion
        onPress={() => generateThumbnail('https://d23dyxeqlo5psv.cloudfront.net/big_buck_bunny.mp4')}
```

File: apps/native-component-list/src/screens/VideoThumbnailsScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        title="Check Valid Source"
+      />
+      <Button
+        onPress={() =>
+          generateThumbnail('http://d23dyxeqlo5psv.cloudfront.net/big_buck_bunny2.mp4')
```
Comment (tsapeta) [inline apps/native-component-list/src/screens/VideoThumbnailsScreen.tsx:29]:
```suggestion
          generateThumbnail('https://d23dyxeqlo5psv.cloudfront.net/big_buck_bunny2.mp4')
```

File: packages/expo-video-thumbnails/CHANGELOG.md [hunk 5]
```diff
@@ -8,6 +8,8 @@
 
 ### 🐛 Bug fixes
 
+[expo-video-thumbnails][Android] fix: crash on bad uri ([#28961](https://github.com/expo/expo/pull/28961) by [@mrakesh0608](https://github.com/mrakesh0608))
```
Comment (tsapeta) [inline packages/expo-video-thumbnails/CHANGELOG.md:11]:
```suggestion
- [Android] Fixed a crash when the source cannot be retrieved. ([#28961](https://github.com/expo/expo/pull/28961) by [@mrakesh0608](https://github.com/mrakesh0608))
```

File: apps/native-component-list/src/navigation/ExpoComponentsStackNavigator.tsx (hunk 1 above)
Comment (mrakesh0608) [reply apps/native-component-list/src/navigation/ExpoComponentsStackNavigator.tsx]: Done

**Review by tsapeta — APPROVED** [review]
Looks good, thanks for the contribution!

**Comment by markrickert** [comment]
Do we know when this fix is going to be released to npm? I've patched my project's version of `8.0.0` with this fix for now but it would be nice to get this fix out there. Thanks!

**Comment by cristian0791** [comment]
> Do we know when this fix is going to be released to npm? I've patched my project's version of `8.0.0` with this fix for now but it would be nice to get this fix out there. Thanks!

Hi @markrickert,
Could you please share the patch? It would be very helpful.
Thanks!

**Comment by markrickert** [comment]
@cristian0791 I don't have access to that client's source code anymore, but it should be relatively easy to make your own patch. just edit the two relevant files from this diff inside your node modules folder and run `npx patch-package expo-video-thumbnails` and commit the patch to git.

**Pull request merged**